giskard/utils/__init__.py

```diff
@@ -21,19 +23,23 @@ class WorkerPool:
 
     def __init__(self):
         self.pool = None
+        self.nb_cancellable = 0
+        self.max_workers = 0
 
-    def start(self, *args, **kwargs):
+    def start(self, max_workers: int = None):
         if self.pool is not None:
             return
         LOGGER.info("Starting worker pool...")
-        self.pool = ProcessPoolExecutor(*args, **kwargs)
+        self.max_workers = min(max_workers, 2) if max_workers is not None else os.cpu_count()
```

Comment:
small improvement: it'd be better to move `2` to the `giskard.settings.Settings`. In this case we'd get a possibility to change it from an env variable

Reply:
Done. I also changed `min` to `max` also, since we always want at least `min_workers`